app/models/classroom.rb

```diff
@@ -25,9 +26,37 @@ class Classroom < ApplicationRecord
   has_many :student_classrooms
   has_many :users, through: :student_classrooms
 
-  validates :name, presence: true
+  validates :subject, :group, presence: true
   validate :user_role
 
+  def parse_for_dashboard
+    {
+      id: self.id,
+      subject: self.subject,
+      group: self.group,
+      teacher: self.user.name,
+      color: self.get_color,
+      link: '#'
+    }
+  end
+
+  def get_color
+    case self.subject
+    when 'English'
+      'blue'
+    when 'Maths'
+      'green'
+    when 'Science'
+      'yellow'
+    when 'Geography'
+      'purple'
+    when 'History'
+      'red'
```

Comment:
forgot to point this out
for better readability you can write it as 
```
    when 'Maths'   then 'green'
    when 'Science' then 'yellow'
```
so you have fewer lines
extra points if the `then`s are aligned